### ISSUE TITLE
Handling simple and double quotes in filenames...

### ISSUE THREAD

**Issue by Creteil** [bassmanitram/actions-for-nautilus#63]
Hi,

Is there a way to handle simple and double quotes in filenames, for example :

* « **Bob's "special" file.txt** ».

**Comment by seindal** [comment]
I ran into this same problem immediately.

I've worked around it by using the %u placeholder, and then URL-decoding the filename in my script.

**Comment by bassmanitram** [comment]
I promise a year is NOT my usual response time! But that's a cool workaround! Can you tell me what you are using to URL decode the file name?

**Comment by seindal** [comment]
I have a little perl script to do that which I use in backquotes from a shell script. Very old style unix way :-)

```
#!/usr/bin/perl -w
use URI::Escape;
if (@ARGV) {
    for (@ARGV) {
	print uri_unescape($_), "\n";
    }
} else {
    while (<>) {
	print uri_unescape($_);
    }
}
```

**Comment by bassmanitram** [comment]
A proper fix is coming!!!! 044db3c

**Comment by Creteil** [comment]
Does a git clone now already reflect the changes ?

**Comment by bassmanitram** [comment]
its all in the feature/strict-match branch at the moment. its stii in
development so expect instability

**Comment by flow-c** [comment]
I've run a little test, and the following characters are not properly escaped in the %f, %b, etc. placeholders:
?
>
<
"
'
;
|
)
(
*
&
`

**Comment by bassmanitram** [comment]
They aren't supposed to be - you have to quote your arguments yourself now
(see new doc)

**Comment by flow-c** [comment]
> They aren't supposed to be - you have to quote your arguments yourself **_now_**

"Now" as in on the `strict-match` branch? Because on `main` only spaces are escaped.

For example with `TEST FILE (Copy).txt`

`%b` (unquoted) results in `TEST\ FILE\ (Copy).txt`,

whereas `"%b"` leads to: `"TEST\ FILE \ (Copy).txt"`.

Neither represent an existing file when interpreted by the shell.

**Comment by bassmanitram** [comment]
Yes... I'm still in the process of finalising how to do the improved
interpolation

**Comment by bassmanitram** [comment]
OK, given that I AM going to have to some in-depth quoting/escaping, the
above set is very helpful - thank you!

**Comment by espositofabian** [comment]
I also hope this problem will get sorted out. With this issue, even a simple plain name can't be copied with `echo "%w" | wl-copy` bur require to get parsed and modified somehow to remove the various `\`.

In any case, @bassmanitram I really wanted to thank you for your work. This was one of the few important feature that was missing from Nautilus. Again, thank you!

**Comment by bassmanitram** [comment]
Hey Fabian - firstly thank you for the kind words!

When using original interpolation (the only one available in stable
releases at the moment), you shouldn't have extraneous backslashes in a
quoted value. You may also have more luck without the echo on wl-copy,
since wl-copy accepts a string as its last argument as the value to go on
the clipboard(s) - which then means you wouldn't need the shell

All that said, it is a cr*pshoot sometimes when constructing commands.
Enhanced interpolation _should_ fix all this - it's currently in the
https://github.com/bassmanitram/actions-for-nautilus/tree/feature/strict-match
branch and about 95% OK (that is I have one slight issue I'm working out -
otherwise it's behaving as it should and the doc is complete) - wouldn't
mind people testing that out for me :)

**Comment by espositofabian** [comment]
> When using original interpolation (the only one available in stable releases at the moment), you shouldn't have extraneous backslashes in a quoted value. You may also have more luck without the echo on wl-copy, since wl-copy accepts a string as its last argument as the value to go on the clipboard(s) - which then means you wouldn't need the shell

Honestly tried various solutions but didn't had much luck. I've settled with `echo "%w" | perl -pe 's/\\ / /g' | wl-copy` for the stable release. But I'm very curious about the new branch.

> wouldn't mind people testing that out for me :)

You can count me in! I'll see to build, install and try out the new branch in these days. I'll report any strange behaviour.

And again, I can't say it enough, thanks for the amazing work!

**Comment by bassmanitram** [comment]
V2 pre-release is out now - with a deb package https://github.com/bassmanitram/actions-for-nautilus/releases/tag/v2.0.0.pre1-1

**Comment by seindal** [comment]
Thanks a million. It works wonderfully now, like it should. Almost all my wrapper scripts around the actions are now gone.

**Comment by flow-c** [comment]
I'm still having issues. If placeholders are quoted, e.g. with `$'%b'`, all the special characters from my table above go through okay.

However, spaces are still escaped by AfN with `\ ` which then of course don't work inside quotes. Hence, filenames such as `File Mon&Tue.txt` cannot be easily processed without wrapper scripts.

May I suggest to leave _all_ special characters, including space, unescaped and leave it to the user to quote all placeholders? Then at least singular placeholders would work consistently. 

Or better, pass all placeholders single-quoted (as the help text in the configurator suggests). Then both singular and plural versions should work.

> If you use the %F placeholder then the paths of all selected files are passed on the command line with each one single-quoted. If you use the %f placeholder, then the command is invoked for each file in the selection with %f replaced by the single-quoted full path of that file.

**Comment by flow-c** [comment]
Fantastic! Thanks so much. Great work, @bassmanitram !

**Comment by bassmanitram** [comment]
> Fantastic! Thanks so much. Great work, [@bassmanitram](https://github.com/bassmanitram) !

(I guess you deleted your other comment :D)

**Comment by flow-c** [comment]
> > Fantastic! Thanks so much. Great work, [@bassmanitram](https://github.com/bassmanitram) !
> 
> (I guess you deleted your other comment :D)

I did :) I had run my old tests without properly restarting AfN. Embarrassing 😳
